hanjaelee/app.js

```diff
@@ -52,20 +53,172 @@ app.post('/signup', (req, res) => {
+    if (rawData['affectedRows']) {
+      return res.status(201).json({ message: 'postCreated' })
+    }
+    return res.status(202).json({ message: 'postnotcreated' })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 4 모든 게시물 가져오기
+app.get('/posts', async (req, res) => {
+  const rawQuery = `
+    SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.image_url AS postingImageUrl,
+      p.content AS postingContent
+    FROM posts AS p
+    INNER JOIN users AS u
+    ON p.user_id = u.id
+    `
+  try {
+    await database.query(rawQuery, (err, rows) => {
+      const result = {}
+      const data = rows
+      result['data'] = data
+      res.status(200).json(result)
     })
-    .catch((error) => {
-      res.status(400).json({ message: error.sqlMessage })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 5 유저 한명이 작성한 모든 게시물 가져오기
+app.get('/posts/:userId', async (req, res) => {
+  const { userId } = req.params
+  const rawQuery = `
+    SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      JSON_ARRAYAGG(JSON_OBJECT(
+                    'postingId', p.id,
+                    'postingImageUrl', p.image_url,
+                    'postingContent', p.content)) AS posting
+    FROM posts AS p
+    INNER JOIN users AS u
+    ON p.user_id = u.id WHERE u.id = ${userId}
+    GROUP BY u.id;`
+
+  try {
+    await database.query(rawQuery, (err, rows) => {
+      return res.status(200).json({ data: rows })
     })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 6 유저 한명이 작성한 게시글 내용 수정하기
+app.patch('/post', async (req, res) => {
+  const { postId, userId, postContent } = req.body
+  let rawQuery = `
+    UPDATE posts
+    SET content = ?
+    WHERE id = ? AND user_id = ?;`
+
+  try {
+    const rawData = await database.query(rawQuery, [postContent, postId, userId])
+    if (rawData['affectedRows']) {
+
+      rawQuery = `
+      SELECT
+        u.id AS userId,
+        u.name AS userName,
+        p.id AS postingId,
+        p.title AS postingTitle,
+        p.content AS postingContent FROM posts AS p
+      INNER JOIN users AS u
+      ON p.user_id = u.id
+      WHERE u.id = ${userId} AND p.id = ${postId}
+      LIMIT 1;`
+
+      try {
+        await database.query(rawQuery, (err, rows) => {
+          if (rows.length == 0) {
+            return res.status(400).json({ message: 'affected rows is empty' })
+          }
+          return res.status(200).json({ data: rows[0] })
+        })
+      } catch (error) {
+        return res.status(200).json({ message: error.sqlMessage })
+      }
+    }
+    return res.status(400).json({ message: `userId and postId is not matched` })
+  }
+  catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
```

Comment:
한재님 아래 코드로도 실행되는지 확인해주실 수 있나요?
```suggestion
app.patch('/post', async (req, res) => {
  try {
    const { postId, userId, postContent } = req.body
    let rawQuery = `
      UPDATE posts
      SET content = ?
      WHERE id = ? AND user_id = ?;`

    const rawData = await database.query(rawQuery, [postContent, postId, userId])
    
    if ( rawData['affectedRows'] ) {
      rawQuery = `
      SELECT
        u.id AS userId,
        u.name AS userName,
        p.id AS postingId,
        p.title AS postingTitle,
        p.content AS postingContent FROM posts AS p
      INNER JOIN users AS u
      ON p.user_id = u.id
      WHERE u.id = ? AND p.id = ?
      LIMIT 1;`

      const [result] = await database.query(rawQuery)
      return res.status(200).json({ data : result })
    }
  }
  catch (error) {
    console.error('Failed to update user posting', error)
    return res.status(400).json({ message: error.sqlMessage })
  }
})
```

Reply:
실행 됩니다 수정하겠습니다